### PR TITLE
build: add enable_remote_module build flag

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Allows to remove the remote module completely at build time.

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes
Notes: no-notes

### CODE REVIEW

File: filenames.auto.gni [hunk 1]
```diff
@@ -134,20 +134,19 @@ auto_filenames = {
...
     "lib/common/api/shell.js",
-    "lib/common/buffer-utils.ts",
     "lib/common/clipboard-utils.ts",
     "lib/common/crash-reporter.js",
     "lib/common/define-properties.ts",
     "lib/common/electron-binding-setup.ts",
     "lib/common/error-utils.ts",
-    "lib/common/is-promise.ts",
+    "lib/common/remote/buffer-utils.ts",
+    "lib/common/remote/is-promise.ts",
```
[the remark this answers is not in the document]
Comment (miniak) [reply filenames.auto.gni]: that would be great

[the remark this answers is not in the document]
Comment (miniak) [reply filenames.auto.gni]: we should do the same for desktop-capturer. maybe a follow up PR?

[the remark this answers is not in the document]
Comment (miniak) [reply filenames.auto.gni]: @MarshallOfSound do you have any tips how to do this?

File: lib/sandboxed_renderer/api/module-list.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ if (features.isDesktopCapturerEnabled()) {
   })
 }
 
-if (process.isRemoteModuleEnabled) {
+if (features.isRemoteModuleEnabled() && process.isRemoteModuleEnabled) {
   moduleList.push({
     name: 'remote',
     loader: () => require('@electron/internal/renderer/api/remote')
```
Comment (nornagon) [inline lib/sandboxed_renderer/api/module-list.ts]: here too

**Comment by miniak** [comment]
@codebytere that’s the plan to do it in a follow ip PR as it should also be done for desktopCapturer, which can also be disabled at build time

**Pull request merged**